examples/hacker_news_assets/hacker_news_assets/__init__.py

```diff
@@ -1 +1,16 @@
-from .repo import hacker_news_assets_local, hacker_news_assets_prod, hacker_news_assets_staging
+from hacker_news_assets.activity_analytics import activity_analytics_definitions
+from hacker_news_assets.core import core_definitions
+from hacker_news_assets.recommender import recommender_definitions
+from hacker_news_assets.resources import get_resource_set_for_deployment
+
+from dagster import repository
+
+
+@repository
+def repo(deployment_name: str):
+    return [
+        core_definitions,
+        recommender_definitions,
+        activity_analytics_definitions,
+        get_resource_set_for_deployment(deployment_name),
+    ]
```

Comment:
Yeah - my initial thinking was to include it as one of the decorator args. I.e.
```
@repository(resource_defs=...)
def my_repo():
    ...
```

but that falls apart if we want the choice of resources to depend on the `deployment_name`. Another option could be:
```
@repository
def my_repo():
    return RepositoryContents(
        jobs_and_assets=[
            core_definitions,
            recommender_definitions,
            activity_analytics_definitions,
            get_resource_set_for_deployment(deployment_name),
        ],
        resource_defs=...,
    )
```

Reply:
I think I'm still confused about what role `ResourceSet` is playing here... could you just have a flat list of resources here?

Reply:
The resources need to be attached to keys. The thinking is that ResourceSet would wrap a dict.

Reply:
maybe something where you pass the definitions through the `ResourceSet`

```
    return    get_resource_set_for_deployment(deployment_name).provide_to([
        core_definitions,
        recommender_definitions,
        activity_analytics_definitions,
    ])
 ```
 
thats not the most elegant though. 
 
In the spirit of brainstorming/exploring If we opened up the design space to class artifacts instead of just decorated functions, we could go all object-oriented-crazy 

```
class DataTeamStuff(DeploymentBasedRepository):
     @staticmethod
     def get_definitions(_deployment_name):
         return [
               core_definitions,
               recommender_definitions,
               activity_analytics_definitions,
          ]
          
      @staticmethod
      def default_resources(deployment_name):
          return {
              "local": ResourceSet(RESOURCES_LOCAL),
              "staging": ResourceSet(RESOURCES_STAGING),
              "prod": ResourceSet(RESOURCES_PROD),
          }[deployment_name]
 ```

Reply:
Kind of a continuation of a previous conversation, but I think if you had `ResourceDefinition` have string keys on the definition itself, you wouldn't need the dict or the resource set.  You would need some extra validation code to check for collisions and a way to alias them though.